packages/clerk-js/src/ui.retheme/elements/SocialButtons.tsx

```diff
@@ -1,14 +1,49 @@
 import type { OAuthProvider, OAuthStrategy, Web3Provider, Web3Strategy } from '@clerk/types';
-import React from 'react';
+import type { Ref } from 'react';
+import React, { forwardRef, isValidElement, useLayoutEffect, useRef, useState } from 'react';
 
-import { Button, descriptors, Grid, Image, localizationKeys, useAppearance } from '../customizables';
-import { useEnabledThirdPartyProviders } from '../hooks';
+import type { LocalizationKey } from '../customizables';
+import {
+  Button,
+  descriptors,
+  Flex,
+  Grid,
+  Icon,
+  Image,
+  localizationKeys,
+  SimpleButton,
+  Spinner,
+  Text,
+  useAppearance,
+} from '../customizables';
+import { useEnabledThirdPartyProviders, useResizeObserver } from '../hooks';
 import type { PropsOfComponent } from '../styledSystem';
 import { sleep } from '../utils';
-import { ArrowBlockButton } from './ArrowBlockButton';
 import { useCardState } from './contexts';
 
 const SOCIAL_BUTTON_BLOCK_THRESHOLD = 2;
+const SOCIAL_BUTTON_PRE_TEXT_THRESHOLD = 1;
+const MAX_STRATEGIES_PER_ROW = 6;
+
+function distributeStrategiesIntoRows<T>(items: T[]): T[][] {
```

Comment:
@marcelscruz Can we move this function inside a utils file or directory and then write some basic tests for it? Or at least explain the functionality in a comment?